peer/peer_test.go

```diff
@@ -0,0 +1,144 @@
+			testLevel: credentials.PrivacyAndIntegrity,
+			want:      true,
+		},
+		{
+			authLevel: credentials.IntegrityOnly,
+			testLevel: credentials.PrivacyAndIntegrity,
+			want:      false,
+		},
+		{
+			authLevel: credentials.IntegrityOnly,
+			testLevel: credentials.NoSecurity,
+			want:      true,
+		},
+		{
+			authLevel: credentials.InvalidSecurityLevel,
+			testLevel: credentials.IntegrityOnly,
+			want:      true,
+		},
+		{
+			authLevel: credentials.InvalidSecurityLevel,
+			testLevel: credentials.PrivacyAndIntegrity,
+			want:      true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.authLevel.String()+"-"+tc.testLevel.String(), func(t *testing.T) {
+			ctx := NewContext(context.Background(), &Peer{AuthInfo: testAuthInfo{credentials.CommonAuthInfo{SecurityLevel: tc.authLevel}}})
+			p, ok := FromContext(ctx)
+			if !ok {
+				t.Fatalf("Unable to get peer from context")
+			}
+			err := credentials.CheckSecurityLevel(p.AuthInfo, tc.testLevel)
+			if tc.want && (err != nil) {
+				t.Fatalf("CheckSeurityLevel(%s, %s) returned failure but want success", tc.authLevel.String(), tc.testLevel.String())
+			} else if !tc.want && (err == nil) {
+				t.Fatalf("CheckSeurityLevel(%s, %s) returned success but want failure", tc.authLevel.String(), tc.testLevel.String())
+			}
+		})
+	}
+}
+
+type addr struct {
+	ipAddress string
+}
+
+func (addr) Network() string   { return "" }
+func (a *addr) String() string { return a.ipAddress }
+
+func TestPeerStringer(t *testing.T) {
+	testCases := []struct {
+		peer *Peer
+		want string
+	}{
+		{
+			peer: &Peer{Addr: &addr{"example.com:1234"}, AuthInfo: testAuthInfo{credentials.CommonAuthInfo{SecurityLevel: credentials.PrivacyAndIntegrity}}},
+			want: "Peer{Addr: 'example.com:1234', LocalAddr: <nil>, AuthInfo: 'testAuthInfo-3'}",
+		},
+		{
+			peer: &Peer{Addr: &addr{"example.com:1234"}, LocalAddr: &addr{"example.com:1234"}, AuthInfo: testAuthInfo{credentials.CommonAuthInfo{SecurityLevel: credentials.PrivacyAndIntegrity}}},
+			want: "Peer{Addr: 'example.com:1234', LocalAddr: 'example.com:1234', AuthInfo: 'testAuthInfo-3'}",
+		},
+		{
+			peer: &Peer{Addr: &addr{"1.2.3.4:1234"}, AuthInfo: testAuthInfo{credentials.CommonAuthInfo{}}},
+			want: "Peer{Addr: '1.2.3.4:1234', LocalAddr: <nil>, AuthInfo: 'testAuthInfo-0'}",
+		},
+		{
+			peer: &Peer{AuthInfo: testAuthInfo{}},
+			want: "Peer{Addr: <nil>, LocalAddr: <nil>, AuthInfo: 'testAuthInfo-0'}",
+		},
+		{
+			peer: &Peer{},
+			want: "Peer{Addr: <nil>, LocalAddr: <nil>, AuthInfo: <nil>}",
+		},
+		{
+			peer: nil,
+			want: "Peer<nil>",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(strings.ReplaceAll(tc.want, " ", ""), func(t *testing.T) {
+			ctx := NewContext(context.Background(), tc.peer)
+			p, ok := FromContext(ctx)
+			if !ok {
+				t.Fatalf("Unable to get peer from context")
+			}
+			if p.String() != tc.want {
+				t.Fatalf("Error using peer String(): expected %q, got %q", tc.want, p.String())
+			}
```

Comment:
```suggestion
			if got := p.String(); got != tc.want {
				t.Fatalf("Unexpected stringer output, got %q; want: %q", got, tc.want)
			}
```

per golang styleguide, got before want always. see: https://google.github.io/styleguide/go/decisions#got-before-want

Reply:
TIL 👍🏻